src/Aspire.Hosting.Kubernetes/api/Aspire.Hosting.Kubernetes.cs

```diff
@@ -0,0 +1,2391 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+namespace Aspire.Hosting.Kubernetes
+{
+    public static partial class KubernetesPublisherExtensions
+    {
+        public static void AddKubernetes(this IDistributedApplicationBuilder builder, string name, System.Action<KubernetesPublisherOptions>? configureOptions = null) { }
+    }
+
+    public sealed partial class KubernetesPublisherOptions : Publishing.PublishingOptions
+    {
+    }
+}
+
+namespace Aspire.Hosting.Kubernetes.Resources
+{
+    [YamlDotNet.Serialization.YamlSerializable]
+    public sealed partial class AffinityV1
```

Comment:
@davidfowl @mitchdenny - do all these classes need to be public? Seems pretty large.

Reply:
This whole package should be pre-release. I think we will iterate and figure out what is needed. For now I consider this entire API surface sus until then 😄

Reply:
Yep - it is prerelease -

https://github.com/dotnet/aspire/blob/b5752c261dcc363cd78980057ed72785a7119390/src/Aspire.Hosting.Kubernetes/Aspire.Hosting.Kubernetes.csproj#L6